nox/sessions.py

```diff
@@ -184,6 +185,13 @@ def create_tmp(self) -> str:
         self.env["TMPDIR"] = tmpdir
         return tmpdir
 
+    @property
+    def cache_dir(self) -> pathlib.Path:
+        """Create and return a 'shared cache' directory to be used across sessions."""
+        path = os.path.join(self._runner.envdir, ".shared")
```

Comment:
So the thing I had in mind for this feature would be `.nox/shared` not `.nox/unit-39/.shared` (and so putting it on `Session` is somewhat of the wrong fit for this).

In https://github.com/dhermes/bezier/tree/2021.2.12, I introduce two artificial directories `.nox/libbezier-debug/` and `.nox/libbezier-release/` and I just point CMake at them for doing builds / rebuilds of the C shared library. Then when the **Python package** gets built (pointing at `libbezier`: https://github.com/dhermes/bezier/blob/2021.2.12/setup.py#L147), I can just re-use the C shared library directory which is **SHARED** by many Nox sessions: `unit-3.7`, `unit-3.8`, `unit-3.9`, `cover`, `functional-3.9`, etc.

```
$ tree -L 2 .nox/libbezier-debug/
.nox/libbezier-debug/
├── build
│   ├── BezierConfig.cmake
│   ├── CMakeCache.txt
│   ├── CMakeFiles
│   ├── Makefile
│   ├── cmake_install.cmake
│   ├── curve.mod
│   ├── curve_intersection.mod
│   ├── helpers.mod
│   ├── install_manifest.txt
│   ├── libbezier.2021.2.12.dylib
│   ├── libbezier.2021.dylib -> libbezier.2021.2.12.dylib
│   ├── libbezier.dylib -> libbezier.2021.dylib
│   ├── quadpack.mod
│   ├── status.mod
│   ├── triangle.mod
│   ├── triangle_intersection.mod
│   └── types.mod
└── usr
    ├── include
    ├── lib
    └── share

6 directories, 16 files
```

Reply:
cc @theacodes

Reply:
So, the creation of `.nox/.shared/` should happen outside of `Session`? Also, I recognize it should not be `.nox/unit-39/.shared`.

Reply:
Where should we do this?

Reply:
Thinking a bit more about this, it probably **DOES** make sense to implement on `Session`, just need the path to refer to `.nox/shared/`, I think the root `/path/to/.nox` is available on the `Session` right?

Reply:
So, the current behavior is OK?

Reply:
Yup, my bad